x-pack/metricbeat/module/aws/utils.go

```diff
@@ -16,14 +16,10 @@ import (
 )
 
 // GetStartTimeEndTime function uses durationString to create startTime and endTime for queries.
-func GetStartTimeEndTime(durationString string) (startTime time.Time, endTime time.Time, err error) {
+func GetStartTimeEndTime(duration time.Duration) (startTime time.Time, endTime time.Time) {
 	endTime = time.Now()
-	duration, err := time.ParseDuration(durationString)
-	if err != nil {
-		return
-	}
-	startTime = endTime.Add(duration)
-	return startTime, endTime, nil
+	startTime = endTime.Add(duration * -2)
```

Comment:
Why `* -2`?

Reply:
We have to set startTime prior to endTime by double the period duration in order to make sure GetMetricDataRequest gets the latest data point. I will add a comment in the code to explain as well.